### ISSUE TITLE
`omitBackground: true` results in screenshots with black background in non-headless mode

### ISSUE THREAD

**Issue by ChALkeR** [puppeteer/puppeteer#2035]
### Steps to reproduce

* Puppeteer version: 1.0.0
* Platform / OS version: Arch Linux
* Node.js version: 9.5.0

**What steps will reproduce the problem?**

Here is the testcase for `about:blank`, though it could be reproduced on any page with transparent background.

```js
const puppeteer = require('puppeteer');
(async () => {
  const browser = await puppeteer.launch({ headless: false });
  const page = await browser.newPage();
  await page.goto('about:blank');
  await page.screenshot({
    clip: { x: 0, y: 0, width: 200, height: 200 },
    omitBackground: true,
    path: 'example.png'
  });
  await browser.close();
})().catch(console.error);
```

1. Launch the provided code snippet, open `example.png`, observe a square filled with pure black (`#000`).
2. Modify the provided code snippet to include `headless: true` instead of `headless: false`, launch, open `example.png`, observe a fully transparent square.
3. Repeat with any other url with transparent color.

**What is the expected result?**

`page.screenshot` should always produce transparent background with `omitBackground: true,` both in headless and non-headless modes.

Alternatively (if the behaviour is intended or wontfix), this should be documented (and perhaps the user should be warned on using both `headless: false` and `omitBackground: true`).

**What happens instead?**

When `headless: false` is set in options, the images rendered with `page.screenshot` always have black background instead of a transparent one in `omitBackground: true` mode. No warnings are produced.

**Comment by aslushnikov** [comment]
@ChALkeR I can't reproduce - the screenshot is  transparent to me. However, when viewed in chrome, the image preview sets the background color to black, and since the screenshot is transparent, it looks as black.

**Comment by ChALkeR** [comment]
@aslushnikov I don't think that's the case here.
I checked those images, one is black (as in not transparent), the other one is transparent (not black).

```console
$ convert example.false.png -format %c histogram:info:-
     40000: (  0,  0,  0) #000000 black
$ convert example.true.png -format %c histogram:info:-
     40000: (  0,  0,  0,  0) #00000000 none
````

`headless: false`:
![example false](https://user-images.githubusercontent.com/291301/36526347-f9d7855c-17be-11e8-98a7-4d38d3a27a4b.png)

`headless: true`:
![example true](https://user-images.githubusercontent.com/291301/36526348-fa122f9a-17be-11e8-9c0b-edb93a3a9e82.png)

I think this should be reopened.

**Comment by aslushnikov** [comment]
@ChALkeR I can't repro on both Linux and OS X with your script using [tip-of-tree](https://github.com/GoogleChrome/puppeteer/commit/b07e705dc9538250757724f2dc6e62f77914b252) PPTR.

Can you try on tip-of-tree and check if it works for you?

**Comment by ChALkeR** [comment]
@aslushnikov Yes, I can confirm that both `@1.1.0` and `@next` (`@1.1.0-next.1519424108630`) are not affected.

This was an issue only on 1.0.0, and it is already fixed by a new release (1.1.0) since the original report date :-).

So, this is resolved already, thanks!

**Comment by alololox** [comment]
This seems to have crept back into version 1.2.0.

I downgraded to 1.1.0 and am not seeing the issue.
